In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px

#### Initializing the parameters that we are going to use in our demo

In [16]:
num_inputs = 3
w = np.random.randn(num_inputs)
x = np.random.rand(num_inputs)
bias = 0.1
target =1

#### Calculating the weighted sum of the neurone

In [17]:
z_1 = np.dot(w, x) + bias
print(z_1)

1.8162580525876122


#### Calculating the function of activation using Sigmoid function and calculating the loss function using cross entroy Loss (both for classification purposes)

In [18]:
y_1 = 1 / (1 + np.exp(-z_1))
loss = -target * np.log(y_1) - (1 - target) * np.log(1 - y_1)

print("Output:", y_1)
print("Loss:", loss)

Output: 0.8601165169317778
Loss: 0.15068741410742803


Output = 0.8601: This means the model predicts an 86.01% probability that the input belongs to the positive class (e.g., class 1).

Confidence: The model is confidently predicting the positive class (close to 1).

Low Loss: A loss of 0.1507 indicates the prediction (0.8601) is close to the target (1), so the model is performing well for this input.

Training Implications: Small gradients during backpropagation (weights/bias won’t change drastically).

The model is already making a reasonable prediction.

### !! Keep in mind that this is only one neurone calculating random numbers and by adding multiple neurones and multiple layers the results will greatly differs and also the interpretations